users/utils.py

```diff
@@ -37,3 +42,53 @@ def user_has_change_permission(user, *models):
 
 def user_has_delete_permission(user, *models):
     return _user_has_models_perms("delete", models)(user)
+
+
+def is_customer(user):
+    if user.is_staff or user.is_superuser:
+        return False
+    elif (
+        user.groups.count() == 1
+        and user.groups.first().name == settings.CUSTOMER_GROUP_NAME
+    ):
+        return True
+    else:
+        return False
+
+
+_berth_customers_group = None
+
+
+def get_berth_customers_group():
+    global _berth_customers_group
+    if not _berth_customers_group:
+        _berth_customers_group = Group.objects.get(name=settings.CUSTOMER_GROUP_NAME)
+    return _berth_customers_group
+
+
+def get_node_user(node):
+    """
+       Try to get to the user this node belongs to
+       either directly by node.user or through CustomerProfile,
+       i.e. by doing node.customer.user
+    """
+    if not node:
+        return None
+
+    node_user = None
+    if hasattr(node, "user"):
+        node_user = node.user
+    elif hasattr(node, "customer") and hasattr(node.customer, "user"):
+        node_user = node.customer.user
+    return node_user
+
+
+def user_is_linked_to_node(user, node):
+    if user is None:
+        return False
+    return user == get_node_user(node)
```

Comment:
Shouldn't it be the other way around tho? I'm probably missing something, but our nodes would be connected to the customer rather than the user, wouldn't they?

I'm thinking, for example, of the berth lease case. They have a `customer` which is a `CustomerProfile`, not a user. But maybe I'm missing something on how graphene/relay handle all this?

Reply:
Resolved: the customers are connected to users, and the authorization is done based on the user, so we check that the logged user is the same as the customer's user.